cmd/api/main.go

```diff
@@ -49,9 +53,23 @@ func run() error {
 	//create and init http server:
 	server := bookhttp.NewServer(bookhttp.ServerConfig{Port: 8080}, bookHandler)
 
-	err = server.ListenAndServe()
-	if err != nil && !errors.Is(err, http.ErrServerClosed) {
-		return fmt.Errorf("unexpected http server error: %w", err)
+	go func() (err error) {
+		err = server.ListenAndServe()
+		if err != nil && !errors.Is(err, http.ErrServerClosed) {
+			return fmt.Errorf("unexpected http server error: %w", err)
+		}
+		return nil
+	}()
+
+	sc := make(chan os.Signal, 1)
+	signal.Notify(sc, syscall.SIGINT, syscall.SIGTERM)
+	<-sc
+
+	ctx, shutdownRelease := context.WithTimeout(context.Background(), 10*time.Second) //HOW MUCH TIME MAKE SENSE FOR US??
```

Comment:
This timeout value should be configurable and we should read it from an environment variable. We also should have an default value, that could be 10 seconds.

Reply:
Ok. We can talk this afternoon if you want. Just tell me at whatsapp.